runtime/common/location.go

```diff
@@ -88,6 +90,21 @@ func NewLocationID(parts ...string) LocationID {
 	return LocationID(strings.Join(parts, "."))
 }
 
+func NewMeteredLocationID(memoryGauge MemoryGauge, parts ...string) LocationID {
+	jointString := joinStrings(memoryGauge, parts)
+	return LocationID(jointString)
+}
+
+func joinStrings(memoryGauge MemoryGauge, parts []string) string {
+	l := 0
+	for _, part := range parts {
+		l += len(part) + 1
+	}
```

Comment:
This will overcount by one. Maybe this approach?

```suggestion
	if len(parts) == 0 {
		return ""
	}
	l := len(parts) - 1 // count the join character "."
	for _, part := range parts {
		l += len(part)
	}
```

Reply:
Yeah, I opted for slight overestimation (1 is very trivial), over complexity